common/models/wordbreakers/src/default/index.ts

```diff
@@ -340,13 +561,29 @@ namespace wordBreakers {
    * Note that
    * @param character a scalar value
    */
-  function property(character: string): WordBreakProperty {
+  function property(character: string, options?: DefaultWordBreakerOptions): WordBreakProperty {
+    // If there is a customized mapping for the character, prioritize that.
+    if(options?.propertyMapping) {
+      let propName = options.propertyMapping(character);
+      if(propName) {
+        return propertyVal(propName, options);
+      }
+    }
+
     // This MUST be a scalar value.
     // TODO: remove dependence on character.codepointAt()?
     let codepoint = character.codePointAt(0) as number;
+
     return searchForProperty(codepoint, 0, WORD_BREAK_PROPERTY.length - 1);
   }
 
+  function propertyVal(propName: string, options?: DefaultWordBreakerOptions) {
+    const matcher = (name: string) => name.toLowerCase() == propName.toLowerCase()
+
+    const customIndex = options?.customProperties?.findIndex(matcher) ?? -1;
+    return customIndex != -1 ? -customIndex - 1 : data.propertyMap.findIndex(matcher);
+  }
+
```

Comment:
After some further thought, it may be possible to drop the `customProperties` array and dynamically build it as custom properties are referenced by this function.  

1.  Check if the property name is not in the default list.  If so, return it.
2.  If not, check if it's already stored within `customProperties` (dynamically built when run).
3.  If stored within `customProperties` already, return the value as above.
4.  If not, add it to the array and return the corresponding value for the new entry.

Of course, this won't help if typos are made in the property name... but there wasn't an existing "catch" for that to begin with, so oh well.

It would make things a little more complex within this function, but it would remove the need for the verbose external boilerplate - the explicit definition of `customProperties`.  (Even when writing it, I thought "this feels like it should be implicit.")

Reply:
Hmm... but trying to infer does run into an issue:  early custom rules may not try to match against the custom property, so it's possible for characters to be classified as "Other" before the new property can be inferred.

Suppose the following:

```typescript
if(context.propertyMatch(null, ["ALetter"], ["Other"], null)) {
  // ...
} else if(context.propertyMatch(null, ["Other"], ["Custom"], null) {
  // ...
}
```

_**Both**_ instances of `["Other"]` could be matched to characters that should be mapped to `["Custom"]` when relying on run-time inference, as `"Custom"` would only be inferred - and thus, defined - when the `["Custom"]` part is processed.  That's... not good.

So... it may be best to rely on an explicit declaration after all, since that can easily avoid such an issue.